### PR TITLE
Bug 1274083, added Configuring Node Resources section

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1274083

@adellape PTAL at this first draft

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@adellape, any first thoughts?

File: admin_guide/manage_nodes.adoc [hunk 1]
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
 Alternatively, instead of specifying specific node names (e.g., `_<node1>_
 _<node2>_`), you can use the `--selector=_<node_selector>_` option to evacuate
 pods on selected nodes.
+
+== Configuring Node Resources
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Add a unique anchor.

File: admin_guide/manage_nodes.adoc [hunk 2]
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
...
+Edit your node configuration file by passing kubelet arguments:
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Maybe: "You can configure node resources by adding kubelet arguments to the <appropriate section of the node config>."

File: admin_guide/manage_nodes.adoc [hunk 3]
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
...
+
+----
+$ vim /etc/origin/node/node-config
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: This needs root permissions, so switch $ to #.

File: admin_guide/manage_nodes.adoc [hunk 4]
```diff
@@ -270,3 +270,26 @@ $ oadm manage-node <node1> <node2> \
...
+----
+
+Example options include:
+
+* `max-pods=40`: Number of Pods that can run on this Kubelet.
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: s/Pods/pod/

s/Kubelet/kubelet/

**Comment by ahardin-rh** [comment]
@rjhowe completed my first pass at this. PTAL. Thanks!

File: admin_guide/manage_nodes.adoc [hunk 5]
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
 Alternatively, instead of specifying specific node names (e.g., `_<node1>_
 _<node2>_`), you can use the `--selector=_<node_selector>_` option to evacuate
 pods on selected nodes.
+
+[[configuring-node-resources]]
+== Configuring Node Resources
+You can configure node resources by adding kubelet arguments to the node
+configuration file:
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Sorry my last comment on this spot got kinda messed up cuz I didn't wrap em in backticks. Basically we need to verify where in this file (e.g., what section/stanza, what parameter name) these should be set.

File: admin_guide/manage_nodes.adoc [hunk 6]
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
...
+
+----
+# vim /etc/origin/node/node-config
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Needs `.yaml` at the end.

File: admin_guide/manage_nodes.adoc [hunk 7]
```diff
@@ -270,3 +270,28 @@ $ oadm manage-node <node1> <node2> \
...
+
+Example options include:
+
+* `max-pods=40`: Number of pods that can run on this kubelet.
+
+* `image-gc-high-threshold=90`: The percent of disk usage after which image garbage
+collection is always run. Default: 90%
+
+* `image-gc-low-threshold=80`: The percent of disk usage before which image garbage
+collection is never run. Lowest disk usage to garbage collect to. Default: 80%
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: This could possibly be a table, if you haven't tried it out as one already.

**Comment by rjhowe** [comment]
@ahardin-rh  we really should show how this will look in the node-config.yaml 

Example:
```
....
kubeletArguments:
  max-pods
    - "40"
  resolv-conf
    - "/etc/resolv.conf"
  image-gc-high-threshold:
    - "90"
  image-gc-low-threshold:
    - "80"
...
```

Taking this even further we could reference how this would look in the ansible hosts file. 
```
openshift_node_kubelet_args={'max-pods': ['40'], 'resolv-conf': ['/etc/resolv.conf'],  'image-gc-high-threshold': ['90'], 'image-gc-low-threshold': ['80']}
```

**Comment by ahardin-rh** [comment]
@rjhowe @adellape Thanks! Made some edits.

**Comment by ahardin-rh** [comment]
@rjhowe @adellape Good to go?

File: admin_guide/manage_nodes.adoc [hunk 8]
```diff
@@ -270,3 +270,47 @@ $ oadm manage-node <node1> <node2> \
 Alternatively, instead of specifying specific node names (e.g., `_<node1>_
 _<node2>_`), you can use the `--selector=_<node_selector>_` option to evacuate
 pods on selected nodes.
+
+[[configuring-node-resources]]
+== Configuring Node Resources
+You can configure node resources by adding kubelet arguments to the node
+configuration file (/etc/origin/node/node-config.yaml). Add the
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Mark up the file name.

File: admin_guide/manage_nodes.adoc [hunk 9]
```diff
@@ -270,3 +270,47 @@ $ oadm manage-node <node1> <node2> \
...
+`*kubeletArguments*` section and include any desired options:
+
+====
+----
+kubeletArguments:
+  max-pods <1>
+    - "40"
+  resolv-conf<2>
```
Comment (adellape) [inline admin_guide/manage_nodes.adoc]: Space before the `<2>`.

**Comment by adellape** [comment]
@ahardin-rh Two minor nits, otherwise LGTM.

**Comment by ahardin-rh** [comment]
@adellape Fixed. Thanks!

**Comment by ahardin-rh** [comment]
@rjhowe further edits?

**Comment by ahardin-rh** [comment]
**revhistory note:**

(for Admin Guide --> Managing Nodes)

    Added the link:../admin_guide/manage_nodes.html#configuring-node-resources[Configuring Node Resources] section.

(for Install/Config --> Advanced Install)

    Added `*openshift_node_kubelet_args*` to the link:../install_config/install/advanced_install.html#configuring-ansible[Host Variables table].

**Comment by adellape** [comment]
This had some follow-up, see https://github.com/openshift/openshift-docs/pull/1719 for new revhistory note.

**Comment by adellape** [comment]
[rev_history]
|link:../admin_guide/manage_nodes.html[Managing Nodes]

|Added the link:../admin_guide/manage_nodes.html#configuring-node-resources[Configuring Node Resources] section.
%

**Pull request merged**